express/code/blocks/ax-marquee-dynamic-hero/ax-marquee-dynamic-hero.js

```diff
@@ -0,0 +1,125 @@
+import { getLibs, getIconElementDeprecated } from '../../scripts/utils.js';
+import { createAccessibilityVideoControls } from '../../scripts/utils/media.js';
+import { appendLinkText, getExpressLandingPageType, sendEventToAnalytics } from '../../scripts/instrument.js';
+
+// Constants for better maintainability
+const METADATA_KEYS = {
+  MARQUEE_INJECT_LOGO: 'marquee-inject-logo',
+};
+
+const CLASS_NAMES = {
+  BUTTON_CONTAINER: 'button-container',
+  QUICK_LINK: 'quick-link',
+  BUTTON: 'button',
+  ACCENT: 'accent',
+  EXPRESS_LOGO: 'express-logo',
+  HERO_TOP: 'hero-top',
+  HERO_ANIMATION_OVERLAY: 'hero-animation-overlay',
+  VIDEO_CONTAINER: 'video-container',
+  AX_MARQUEE_DYNAMIC_HERO: 'ax-marquee-dynamic-hero',
+};
+
+const LOGO_INJECT_VALUES = ['on', 'yes'];
+
+let getMetadata;
+
+/**
+ * Adds button styling classes to the CTA link
+ * @param {Element} block - The main block element
+ */
+function setupButtonStyling(block) {
+  const h1 = block.querySelector('h1:first-of-type');
```

Comment:
will this block only work with h1? Otherwise we should add a heading class to like h1,h2,h3